test/consul_helpers.go

```diff
@@ -0,0 +1,196 @@
+
+const GoogleProjectIDEnvVar = "GOOGLE_CLOUD_PROJECT_ID"
+
+// Test the consul-cluster example by:
+//
+// 1. Copying the code in this repo to a temp folder so tests on the Terraform code can run in parallel without the
+//    state files overwriting each other.
+// 2. Building the Image in the consul-image example with the given build name
+// 3. Deploying that Image using the consul-cluster Terraform code
+// 4. Checking that the Consul cluster comes up within a reasonable time period and can respond to requests
+func runConsulClusterTest(t *testing.T, packerBuildName string, examplesFolder string, packerTemplatePath string) {
+	exampleFolder := test_structure.CopyTerraformFolderToTemp(t, RepoRoot, examplesFolder)
+
+	test_structure.RunTestStage(t, "setup_image", func() {
+		// Get the Project Id to use
+		gcpProjectID := gcp.GetGoogleProjectIDFromEnvVar()
+
+		// Pick a random GCP zone to test in. This helps ensure your code works in all regions.
+		gcpZone := gcp.GetRandomZone(t, nil, nil)
+
+		test_structure.SaveString(t, exampleFolder, SavedGCPZone, gcpZone)
+
+		// Make sure the Packer build completes successfully
+		imageID := buildImage(t, packerTemplatePath, packerBuildName, gcpProjectID, gcpZone)
+		test_structure.SaveArtifactID(t, exampleFolder, imageID)
+	})
+
+	defer test_structure.RunTestStage(t, "teardown", func() {
+		terraformOptions := test_structure.LoadTerraformOptions(t, exampleFolder)
+		terraform.Destroy(t, terraformOptions)
+
+		// Get the Project Id to use
+		gcpProjectID := gcp.GetGoogleProjectIDFromEnvVar()
+
+		imageID := test_structure.LoadArtifactID(t, exampleFolder)
+		defer gcp.DeleteImage(t, gcpProjectID, imageID)
+	})
+
+	test_structure.RunTestStage(t, "deploy", func() {
+		// GCP only supports lowercase names for some resources
+		uniqueID := strings.ToLower(random.UniqueId())
+		serverClusterName := fmt.Sprintf("consul-server-cluster-%s", uniqueID)
+		clientClusterName := fmt.Sprintf("consul-client-cluster-%s", uniqueID)
+		gcpProjectID := gcp.GetGoogleProjectIDFromEnvVar()
+		gcpZone := test_structure.LoadString(t, exampleFolder, SavedGCPZone)
+		imageID := test_structure.LoadArtifactID(t, exampleFolder)
+
+		terraformOptions := &terraform.Options{
+			TerraformDir: exampleFolder,
+			Vars: map[string]interface{}{
+				ConsulClusterExampleVarProject:              gcpProjectID,
+				ConsulClusterExampleVarZone:                 gcpZone,
+				ConsulClusterExampleVarServerClusterName:    serverClusterName,
+				ConsulClusterExampleVarClientClusterName:    clientClusterName,
+				ConsulClusterExampleVarServerClusterTagName: serverClusterName,
+				ConsulClusterExampleVarClientClusterTagName: clientClusterName,
+				ConsulClusterExampleVarServerSourceImage:    imageID,
+				ConsulClusterExampleVarClientSourceImage:    imageID,
+				ConsulClusterExampleVarServerClusterSize:    ConsulClusterExampleDefaultNumServers,
+				ConsulClusterExampleVarClientClusterSize:    ConsulClusterExampleDefaultNumClients,
+			},
```

Comment:
These are more stylistic preferences so I'm not too opinionated about it. It just seems a little cleaner to use literals directly instead of `const`s that are only used once.